exchanges/huobi/huobi_websocket.go

```diff
@@ -75,427 +73,233 @@ var subscriptionNames = map[string]string{
 	subscription.CandlesChannel:   wsCandlesChannel,
 	subscription.OrderbookChannel: wsOrderbookChannel,
 	subscription.AllTradesChannel: wsTradesChannel,
-	/* 	TODO: Pending upcoming V2 support, these are dropped from the translation table so that the sub conf will be correct and not need upgrading, but will error on usage
-	subscription.MyTradesChannel:  wsMyOrdersChannel,
-	subscription.MyOrdersChannel:  wsMyTradesChannel,
+	subscription.MyTradesChannel:  wsMyTradesChannel,
+	subscription.MyOrdersChannel:  wsMyOrdersChannel,
 	subscription.MyAccountChannel: wsMyAccountChannel,
-	*/
 }
 
-// Instantiates a communications channel between websocket connections
-var comms = make(chan WsMessage)
-
 // WsConnect initiates a new websocket connection
 func (h *HUOBI) WsConnect() error {
 	if !h.Websocket.IsEnabled() || !h.IsEnabled() {
 		return stream.ErrWebsocketNotEnabled
 	}
-	var dialer websocket.Dialer
-	err := h.wsDial(&dialer)
-	if err != nil {
+	if err := h.Websocket.Conn.Dial(&websocket.Dialer{}, http.Header{}); err != nil {
 		return err
 	}
+	ctx := context.Background()
 
-	if h.Websocket.CanUseAuthenticatedEndpoints() {
-		err = h.wsAuthenticatedDial(&dialer)
-		if err != nil {
-			log.Errorf(log.ExchangeSys,
-				"%v - authenticated dial failed: %v\n",
-				h.Name,
-				err)
-		}
-		err = h.wsLogin(context.TODO())
-		if err != nil {
-			log.Errorf(log.ExchangeSys,
-				"%v - authentication failed: %v\n",
-				h.Name,
-				err)
-			h.Websocket.SetCanUseAuthenticatedEndpoints(false)
-		}
-	}
-
-	h.Websocket.Wg.Add(1)
-	go h.wsReadData()
-	return nil
-}
-
-func (h *HUOBI) wsDial(dialer *websocket.Dialer) error {
-	err := h.Websocket.Conn.Dial(dialer, http.Header{})
-	if err != nil {
-		return err
-	}
 	h.Websocket.Wg.Add(1)
-	go h.wsFunnelConnectionData(h.Websocket.Conn, wsMarketURL)
-	return nil
-}
+	go h.wsReadMsgs(h.Websocket.Conn)
 
-func (h *HUOBI) wsAuthenticatedDial(dialer *websocket.Dialer) error {
-	if !h.IsWebsocketAuthenticationSupported() {
-		return fmt.Errorf("%v AuthenticatedWebsocketAPISupport not enabled",
-			h.Name)
-	}
-	err := h.Websocket.AuthConn.Dial(dialer, http.Header{})
-	if err != nil {
-		return err
+	h.Websocket.SetCanUseAuthenticatedEndpoints(false)
+	if h.IsWebsocketAuthenticationSupported() {
```

Comment:
?
I don't see any other exchange behave in this manner

Reply:
In which sense?
Most of them use `SetCanUseAuthenticatedEndpoints` false on failure (Kraken, Okx)
Many of them use `IsWebsocketAuthenticationSupported` to choose to dial auth (Kraken, Okx)

This is using a different pattern, but I'm actually happier with it.
Set CanUseAuthEndpoints to false on a (re)connect, and then true if the auth succeeds.
It feels cleaner than having the branching logic we've seen elsewhere.
Definitely cleaner than using `CanUseAuthenticatedEndpoints` to decide to even dial, which I saw on one exchange, and feels definitely wrong.

Is there a downside to this that I'm missing?

If anything, I'd follow this pattern with others when I'm interfering with them.

Reply:
Why set something to false when you don't even know if they have credentials set or if its going to be used? 
If you put the `h.Websocket.SetCanUseAuthenticatedEndpoints(false)` inside the if statement, it no longer looks out of place and that change will still work with all the reasoning you outlined

Reply:
> Why set something to false when you don't even know if they have credentials set or if its going to be used?
Because if we aren't about to go into the `if` statement, for any reason at all, then `false` is the correct state for the value. Not setting it to false seems wrong to me; Doing it inside an if statement that _could_ be false and not run seems wrong to me.

I don't know what the use-cases might be. Feels like the user would have to unset credentials for it to need to be set to false where I have it. But that's kind of irrelevant. My point is that it's the best representation of the state at the right time.

Reply:
Kinda afraid you're going to ask me to change all the others if I convince you 🤦

Reply:
After 30 days, I still don't like it, nor would I request you put it anywhere else. I didn't expect to talk this much about a bool placement which doesn't affect performance. Its just doing something before even knowing its necessary.

Reply:
I too don't see the point of this, we should only SetCanUseAuthenticatedEndpoints(false) if auth fails. We'll also always default to off unless a user has provided auth credentials and enabled websocket, which we'll test upon engine starting up

Reply:
Okay; I'll change it.

I'm still sticking to my view that `WsConnect` is the point at which we have *lost* whatever possible authenticated state we had before, not on failed auth response.
( _Arguably we lost it on disconnect_ )
So if reconnect and we don't go into `h.IsWebsocketAuthenticationSupported` for literally any reason now, or in the future, we've left the bool in the wrong state. I know that's an edge case. I just don't see the point in not coding for it.

Fixup coming.

Reply:
Fixed https://github.com/gbjk/gocryptotrader/commit/53425ab42

Moved the set to true up into WsConnect because that matches other exchanges (Kraken), and means setting true and false is in the same place.